### ISSUE TITLE
Failed to install the extension (VS 2017, VS 2019)

### ISSUE THREAD

**Issue by werat** [laurentkempe/GitDiffMargin#414]
## Versions
 - Git Diff Margin version: 3.10.3.462 (latest)
 - Visual Studio version: 2017 Professional (15.9.36), 2019 Professional (16.10.3)

## What happened
Cannot install the extension neither via the extensions manager in Visual Studio, nor by manually running the GitDiffMargin.vsix. The installer says "This extension is not installable on any currently installed products". See the screenshot and the full log below.

I've tried using the `vsixinstaller.exe` from Visual Studio 2019, but the error message and the logs are the same:
```
PS> & "C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\VSIXInstaller.exe" .\GitDiffMargin.vsix
```

I do have Visual Studio 2022 Preview installed, so I suspect it might be the issue. I've used GitDiffMargin before and it worked great! According to the manifest it seems GitDiffMargin should work in VS 2022 Preview too, but I didn't really expect it to.

### Screenshots
![image](https://user-images.githubusercontent.com/241450/124249673-62556000-db24-11eb-8043-6d1660ba1a0c.png)

### Logs
```
2021-07-02 11:01:23 - Microsoft VSIX Installer
2021-07-02 11:01:23 - -------------------------------------------
2021-07-02 11:01:23 - vsixinstaller.exe version:
2021-07-02 11:01:23 - 17.0.1616-preview1
2021-07-02 11:01:23 - -------------------------------------------
2021-07-02 11:01:23 - Command line parameters:
2021-07-02 11:01:23 - C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service\VSIXInstaller.exe,C:\Users\werat\Downloads\GitDiffMargin.vsix
2021-07-02 11:01:23 - -------------------------------------------
2021-07-02 11:01:23 - Microsoft VSIX Installer
2021-07-02 11:01:23 - -------------------------------------------
2021-07-02 11:01:24 - Initializing Install...
2021-07-02 11:01:24 - Extension Details...
2021-07-02 11:01:24 - 	Identifier         : GitDiffMargin..5ef9f816-d9ce-45cd-a8fa-cd67a807df0e
2021-07-02 11:01:24 - 	Name               : Git Diff Margin
2021-07-02 11:01:24 - 	Author             : Laurent Kempé
2021-07-02 11:01:24 - 	Version            : 3.10.3.462
2021-07-02 11:01:24 - 	Description        : Git Diff Margin displays live Git changes of the currently edited file on Visual Studio margin and scroll bar. Supports Visual Studio 2012 through Visual Studio 2019.
2021-07-02 11:01:24 - 	Locale             : en-US
2021-07-02 11:01:24 - 	MoreInfoURL        : https://github.com/laurentkempe/GitDiffMargin
2021-07-02 11:01:24 - 	InstalledByMSI     : False
2021-07-02 11:01:24 - 	SupportedFrameworkVersionRange : [0.0,2147483647.2147483647]
2021-07-02 11:01:24 - 
2021-07-02 11:01:24 - 	SignatureState     : Unsigned
2021-07-02 11:01:24 - 	Supported Products : 
2021-07-02 11:01:24 - 		Microsoft.VisualStudio.Community
2021-07-02 11:01:24 - 			Version : [11.0,18.0)
2021-07-02 11:01:24 - 			ProductArchitecture : notSpecified
2021-07-02 11:01:24 - 		Microsoft.VisualStudio.Pro
2021-07-02 11:01:24 - 			Version : [11.0,18.0)
2021-07-02 11:01:24 - 			ProductArchitecture : notSpecified
2021-07-02 11:01:24 - 
2021-07-02 11:01:24 - 	References         : 
2021-07-02 11:01:24 - 	Prerequisites      : 
2021-07-02 11:01:24 - 		-------------------------------------------------------
2021-07-02 11:01:24 - 		Identifier   : Microsoft.VisualStudio.Component.CoreEditor
2021-07-02 11:01:24 - 		Name         : Visual Studio core editor
2021-07-02 11:01:24 - 		Version      : [15.0,18.0)
2021-07-02 11:01:24 - 
2021-07-02 11:01:24 - Signature Details...
2021-07-02 11:01:24 - 	Extension is not signed.
2021-07-02 11:01:24 - 
2021-07-02 11:01:24 - Searching for applicable products...
2021-07-02 11:01:24 - Found installed product - Global Location
2021-07-02 11:01:24 - Found installed product - Visual Studio Professional 2019
2021-07-02 11:01:24 - Found installed product - Visual Studio Professional 2022 Preview
2021-07-02 11:01:24 - Found installed product - Visual Studio Professional 2017
2021-07-02 11:01:24 - VSIXInstaller.NoApplicableSKUsException: This extension is not installable on any currently installed products.
   at VSIXInstaller.ExtensionService.GetInstallableDataImpl(IInstallableExtension extension, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
   at VSIXInstaller.ExtensionService.GetInstallableData(String vsixPath, String extensionPackParentName, Boolean isRepairSupported, IStateData stateData, IEnumerable`1& skuData)
   at VSIXInstaller.ExtensionPackService.IsExtensionPack(IStateData stateData, Boolean isRepairSupported)
   at VSIXInstaller.ExtensionPackService.ExpandExtensionPackToInstall(IStateData stateData, Boolean isRepairSupported)
   at VSIXInstaller.App.Initialize(Boolean isRepairSupported)
   at VSIXInstaller.App.Initialize()
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)

```

**Comment by laurentkempe** [comment]
Thanks for the report @werat 

It is not working yet on VS 2022 preview 1 but @sharwell is working on it #401. 

I also have VS 2022 preview 1 installed too and I can reproduce the issue. I have uninstalled Git Diff Margin from VS 20219 and tried to re-install it and got the same issue with the same log.

I might have to [revert that change](https://github.com/laurentkempe/GitDiffMargin/commit/cb718faef8fced226cc3aba8131c0d472d70c70d) @sharwell

**Comment by laurentkempe** [comment]
@werat you can download a version which fix the issue from there [build artifact here](https://ci.appveyor.com/api/buildjobs/jnk03oth0gwf4uf5/artifacts/GitDiffMargin.Extension%2Fbin%2FRelease%2FGitDiffMargin.vsix)

I might do a real release later.

**Comment by werat** [comment]
Thanks, this build works!

**Comment by sharwell** [comment]
Interesting I'll take a look

**Comment by sharwell** [comment]
Hi @werat, [this build artifact](https://ci.appveyor.com/project/laurentkempe/gitdiffmargin/builds/39848491/artifacts) should be installable in VS 2019 _and_ also work in VS 2022.

**Comment by sharwell** [comment]
This was fixed by the changes to source.extension.vsixmanifest in 548b527392a6175fc6c3ea9a03a172828e14692b.

**Comment by laurentkempe** [comment]
> [this build artifact](https://ci.appveyor.com/project/laurentkempe/gitdiffmargin/builds/39848491/artifacts) should be installable in VS 2019 _and_ also work in VS 2022.

I tried it and could install it on VS 2022 preview 1 without any issues and it works like a charm @sharwell

**Comment by laurentkempe** [comment]
> This was fixed by the changes to source.extension.vsixmanifest in [548b527](https://github.com/laurentkempe/GitDiffMargin/commit/548b527392a6175fc6c3ea9a03a172828e14692b). The vsixinstaller for dev17 doesn't like the lack of `<ProductArchitecture>`.

I see now, [from the doc](https://docs.microsoft.com/en-us/visualstudio/extensibility/migration/update-visual-studio-extension?view=vs-2022#add-a-visual-studio-2022-target)

> For product versions less than 17.0 the default value is x86 and can be omitted. For product versions 17.0 and greater this element is required and there is no default value. For Visual Studio 2022 the only valid content for this element is "amd64".

I will close this issue as this is now resolved with published [version 3.10.4](https://marketplace.visualstudio.com/items?itemName=LaurentKempe.GitDiffMargin), and it will be enhanced with the changes `[548b527](https://docs.microsoft.com/en-us/visualstudio/extensibility/migration/update-visual-studio-extension?view=vs-2022#add-a-visual-studio-2022-target)` and available when we have #401 merged.
